src/android/MyConnectionService.java

```diff
@@ -16,91 +19,275 @@
+                this.callNotification = new CallNotification(payloadString, context);
+                this.callNotification.show();
+            }
+
+            private void closeNotification() {
+                if (this.callNotification != null) {
+                    this.callNotification.close();
+                    this.callNotification = null;
+                }
+            }
+
             @Override
             public void onAnswer() {
+                Log.d(TAG, "onAnswer()");
+                this.closeNotification();
+
                 this.setActive();
-                // Intent intent = new Intent(CordovaCall.getCordova().getActivity().getApplicationContext(), CordovaCall.getCordova().getActivity().getClass());
-                // // intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK|Intent.FLAG_ACTIVITY_SINGLE_TOP);
-                // intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK|Intent.FLAG_ACTIVITY_SINGLE_TOP|Intent.FLAG_FROM_BACKGROUND);
-                // CordovaCall.getCordova().getActivity().getApplicationContext().startActivity(intent);
-                // ArrayList<CallbackContext> callbackContexts = CordovaCall.getCallbackContexts().get("answer");
-                // for (final CallbackContext callbackContext : callbackContexts) {
-                //     CordovaCall.getCordova().getThreadPool().execute(new Runnable() {
-                //         public void run() {
-                //             Bundle data = request.getExtras() != null ? request.getExtras() : new Bundle();
-                //             PluginResult result = new PluginResult(PluginResult.Status.OK, convertBundleToJson(data));
-                //             result.setKeepCallback(true);
-                //             callbackContext.sendPluginResult(result);
-                //         }
-                //     });
-                // }
-                // TelecomManager tm = (TelecomManager) CordovaCall.getCordova().getActivity().getApplicationContext().getSystemService(Context.TELECOM_SERVICE);
-                // tm.showInCallScreen(false);
+                activeConnectionUUID = callUUID;
+
+                showWebApp("answerCall", payloadString);
 
                 CordovaCall.emitEvent("answer", new PluginResult(PluginResult.Status.OK, payloadString));
             }
 
             @Override
             public void onReject() {
-                DisconnectCause cause = new DisconnectCause(DisconnectCause.REJECTED);
-                this.setDisconnected(cause);
-                this.destroy();
-                conn = null;
+                Log.d(TAG, "onReject, call_uuid: " + callUUID);
+                this.closeNotification();
+
+                disconnectConnection(callUUID, DisconnectCause.REJECTED);
+
+                showWebApp("declineCall", payloadString); // Controversial UX but doing so that we can tell the web app to reject the call (which may let the caller not it was declined)
```

Comment:
Is there a way to start the app without bringing it to the front? If that's possible then it might be better UX.